pyproject.toml

```
@@ -11,11 +11,43 @@ build-backend = "setuptools.build_meta"
 
 [project]
 name = "velocity_modelling"
-authors = [{ name = "QuakeCoRE" }]
+authors = [{ name = "ucgmsim" }]
 description = "Velocity modelling library"
 readme = "README.md"
 requires-python = ">=3.11"
-dynamic = ["version", "dependencies"]
+dynamic = ["version"]
+dependencies = [
+    "numpy",
+    "h5py",
+    "pandas[parquet]",
+    "matplotlib",
+    "shapely",
+    "geopandas",
+    "numba",
+    "pyyaml",
+    "tqdm",
+    "typer",
+    "threadpoolctl",
+    "cartopy",
+    "qcore-utils",
+]
+
+[project.optional-dependencies]
+test = [
+    "hypothesis[numpy]",
+    "pytest",
+    "pytest-cov",
+    "pytest-repeat",
+]
+
+# Configure Deptry to not report these as unused dependencies
+[tool.deptry.per_rule_ignores]
+DEP002 = ["pytest", "pytest-cov", "pytest-repeat", "hypothesis"]
```

Rather than explicitly listing every dependency as a per-rule ignore, you should list all the whole group as a dev dependency group.


```suggestion
[tool.deptry]
pep621_dev_dependency_groups = ["test"]
```

---

This is because new test dependencies will be automatically excluded.